### ISSUE TITLE
Unable to perform right in the preview version 4-22

### ISSUE THREAD

**Issue by taooceros** [nod5/HighlightJump#9]
I am unable to use the annotation tools provided by this plugin in the preview version released at 4-22.
Whenever I tried to annotate with the script, it is asking me to save as the documents. And there seems no change of the page unless the pdf is reopened.

**Comment by nod5** [comment]
Thank you for the bug report. I can reproduce the problem. Will fix. Note though that the annotations feature is currently in active development in SumatraPDF so some things may soon break again if updating to the newer prerelease versions.

Note to myself:
SumatraPDF has changed windows message numbers in resources.h , see https://github.com/sumatrapdfreader/sumatrapdf/commit/f1045f24934c0581a110923b1cd4819114fc0913

**Comment by taooceros** [comment]
Thanks! I am looking forward for the use of this script in sumatrapdf!

**Comment by nod5** [comment]
Ok, I have fixed the issue in HighlightJump. But there is also a bug in the latest SumatraPDF prerelease related to the `A` shortcut and saving highlights that still prevents HighlightJump from working well. 

I will wait for that other bug to be fixed first. If you want to use HighlightJump in the meantime you can use it with earlier SumatraPDF prereleases. Version ~~SumatraPDF-prerelease-12280-64-install.exe~~ **edit:** SumatraPDF-prerelease-12145-64.exe  works for me.

https://github.com/sumatrapdfreader/sumatrapdf/issues/1581

**Comment by GitHubRulesOK** [comment]
@nod5 
Note there is a limit to older versions <strike>thus an official copy of daily 12280 is not available </strike>
latest daily cull was

> 12492, deleting
>   software/sumatrapdf/daily/SumatraPDF-prerel-12492-64-install.exe deleting
> 

It could be a problem if you host your own out of date copies but as considerd before when suggesting your own compiler mods, it may be worth you keeping a working "reference" copy one step back for occasions when there is a significant shift in code.

**Comment by nod5** [comment]
@GitHubRulesOK Yes, I won't host the files. I mentioned it mostly in case people reading this has kept a copy of the old version. That said it seems at least some older prerelease versions are still accessible by taking the URL for the newest prerelease and replacing the filename part with that of the old prerelease. In this case: https://kjkpubsf.sfo2.digitaloceanspaces.com/software/sumatrapdf/prerel/SumatraPDF-prerelease-12280-64-install.exe
The download has SHA1 hash 5044b378c53ca984fdaea006f07db3b7b2e9be9f which matches my old download of that prerelease back in march.

**Comment by GitHubRulesOK** [comment]
Ok I did not spot that one as still available it was the last pre-rel before formal release 3.2 so was possibly kept for historic reasons but note the daily are deleted on an automated cycle as new dailys are compiled
on testing pre-rel copies a newer one e.g. https://kjkpubsf.sfo2.digitaloceanspaces.com/software/sumatrapdf/prerel/SumatraPDF-prerel-12387-64.exe is also still available so they must be on a different cycle
can you tell which commit makes a difference to your app to save me testing every dozen copies your link above to commit suggests 12218 but that was prior to 12280 and my initial test with 12387 suggests there may be a problem (the one you fixed?) but I don't see the colors disappear in 387 as described for 511?

**Comment by nod5** [comment]
Yeah the file retention seems to differ for these two tracks
https://www.sumatrapdfreader.org/dailybuilds.html
https://www.sumatrapdfreader.org/prerelease.html

> can you tell which commit makes a difference to your app to save me testing every dozen copies your link above to commit suggests 12218 but that was prior to 12280

Apologies, I spoke incorrectly. I had mixed up which version I was testing what with. 

Correct information: `SumatraPDF-prerelease-12145-64.exe` works with HighlightJump version `2020-03-01`. But that prerelease does not seem available on official link anymore. SHA1 hash is `f3b82f8c298545414fb6395d3707c6e202244cc2` .

Note to self: for new HighlightJump releases write in readme what the latest confirmed compatible SumatraPDF prerelease version is at that time.

**Comment by GitHubRulesOK** [comment]
I can confirm it is not working in present 12564 (daily) but I need a keyword related to the change in annotation to at least tell at which version it may have changed

**Comment by nod5** [comment]
I haven't had time to pin down the what and when of the newer issue. But something must have changed after function `MakeAnnotationFromSelection` is called in [SumatraPDF.cpp](https://github.com/sumatrapdfreader/sumatrapdf/blob/master/src/SumatraPDF.cpp) . Also function `OnMenuSaveAnnotationsToSmx`.

**Comment by GitHubRulesOK** [comment]
ok the first word suggests around 12480

**Comment by GitHubRulesOK** [comment]
ok the first word suggests around 12480 and then found it was later showing problems around 12500 so misbehaviour in HighlightJump starts after a fix about 12508

Last version I think was working before bug was probably **SumatraPDF-prerel-12434-install** (32bit etc) can you test with your changed copy to see if it was more stable around that version

prerel IS on a different delete cycle as latest cull is `12242, deleting`
  software/sumatrapdf/prerel/SumatraPDF-prerelease-12242-64-install.exe deleting

unsure if 12280 will be next to go

**Comment by nod5** [comment]
I'm short on time so will just wait until kjk fixes the SumatraPDF bug.
